benchmarks/transport/CMakeLists.txt

```diff
@@ -11,23 +11,11 @@ if (GHEX_USE_UCP)
         add_executable(${_t} ${_t}_mt.cpp )
         target_compile_definitions(${_t} PRIVATE USE_HEAVY_CALLBACKS USE_RAW_SHARED_MESSAGE USE_POOL_ALLOCATOR)
         target_link_libraries(${_t} ghexlib)
-        if (GHEX_USE_PMIX)
-            target_compile_definitions(${_t} PRIVATE USE_PMIX)
-        endif()
-        if (GHEX_USE_UCP)
-            target_compile_definitions(${_t} PRIVATE USE_UCX)
```

Comment:
One CMake way of doing it is to set `target_compile_definitions(${_t} PUBLIC USE_UCX)`. In this way, when you link with GHEX cmake will add these options to the compilation line and your object files. It is true that if these options do not change once the library is installed, then we do not need this and we can use the `config.h`

Reply:
@mbianco `config.h` is regenerated every time you change the cmake input files and run generate, and it is also only created in the build directory (not installed). The point of it is that you don't have to add individual `target_compile_definitions(${_t} PUBLIC ...` for each new config variable you implement. It makes shorter and simpler to maintain `CMakeList.txt` files, but also more compact compile command lines without lots of `-D` statements.